mw-webapp/src/component/table/Table.tsx

```diff
@@ -1,40 +1,28 @@
-import {useEffect, useState} from "react";
-import {
-  flexRender,
-  getCoreRowModel,
-  useReactTable,
-} from "@tanstack/react-table";
-import {columns} from "src/component/table/columns";
-import {getDayReports} from "src/dataAccessLogic/getDayReports";
-import {DayReport} from "src/model/businessModel/DayReport";
+import {memo} from "react";
+import {flexRender} from "@tanstack/react-table";
+import {TableReports} from "src/pages/reportsTable/ReportsTable";
 import styles from "src/component/table/Table.module.scss";
 
 /**
- * Table (need update for split component and logic code)
+ * Table's props
  */
-export const Table = () => {
-  const [data, setData] = useState<DayReport[]>([]);
-
-  const loadDayReports = async () => {
-    const dayReports = await getDayReports();
-    setData(dayReports);
-  };
-
-  useEffect(() => {
-    loadDayReports();
-  }, []);
-
-  const table = useReactTable({
-    data,
-    columns,
-    getCoreRowModel: getCoreRowModel(),
-  });
+interface TableProps {
+  /**
+  * Table's data
+  */
+  data: TableReports;
```

Comment:
TableReports includes DayReport and it makes table component related to business models and not reusable. What do you think about using generics for table? Look like it should be good approach.

Reply:
I haven’t found any other way other than generics and union type. Otherwise, the generic cannot be determined. type TablesModel will be a union type of all models coming into the table